pygmt/tests/test_clib_virtualfile_from_vectors.py

```diff
@@ -53,17 +53,30 @@ def test_virtualfile_from_vectors(dtypes):
 
 
 @pytest.mark.benchmark
-@pytest.mark.parametrize("dtype", [str, object])
-def test_virtualfile_from_vectors_one_string_or_object_column(dtype):
-    """
-    Test passing in one column with string or object dtype into virtual file dataset.
+@pytest.mark.parametrize(
+    ("array_func", "dtype"),
+    [
+        pytest.param(np.array, {"dtype": np.str_}, id="str"),
+        pytest.param(np.array, {"dtype": np.object_}, id="object"),
+        pytest.param(
+            getattr(pa, "array", None),
+            {},  # pa.string()
+            marks=skip_if_no(package="pyarrow"),
+            id="pyarrow",
+        ),
```

Comment:
A possible solution to avoid the complicated param is:
```
try:
    import pyarrow as pa
    pa_array = pa.array
except ImportError:
    pa_array = None

@pytest.mark.parametrize(
    ("array_func", "dtype"),
    [
        pytest.param(np.array, {"dtype": np.str_}, id="str"),
        pytest.param(np.array, {"dtype": np.object_}, id="object"),
        pytest.param(pa_array, {}, marks=skip_if_no(package="pyarrow"), id="pyarrow"),
    ]
)
```

Reply:
Done at 85d2bb510650daf6685311c17f826b27c8436861